app/routes/ui/completed_page.py

```diff
@@ -0,0 +1,42 @@
+import json
+
+from cache.cache import check_and_get_scenario_by_name
+from library.postgres import get_scenario_by_friendly_name_from_postgres
+from models import get_session
+from models.scenario import Scenario
+from routes.base import BaseUIHandler
+
+
+class UIScenarioCompletedPageHandler(BaseUIHandler):
+    """
+    Handle routes that have {scenario_friendly_name}/completed
+    """
+
+    async def get(self, scenario_friendly_name):
+        try:
+            scenario_dict = await check_and_get_scenario_by_name(scenario_friendly_name)
+            if not scenario_dict:
+                if not self.db_session:
+                    self.db_session = get_session()
+                scenario_obj: Scenario = (
+                    await get_scenario_by_friendly_name_from_postgres(
+                        scenario_friendly_name, self.db_session, update_cache=True
+                    )
+                )
+            else:
+                scenario_obj: Scenario = Scenario(**scenario_dict)
+            if not scenario_obj.is_published:
+                raise ValueError("This Scenario is currently not accessible")
+
+            await self.render(
+                "completed_page.html",
+                scenario_name=scenario_obj.name,
+                scenario_friendly_name=scenario_obj.friendly_name,
```

Comment:
`scenario_friendly_name` isn't used in the completion page right?

Reply:
yeah we dont use it. I had left it in case it would have been needed. But ill remove it for now